shared/src/question-generators/huffman-coding/huffManCodingGenerateWords.ts

```diff
@@ -11,120 +11,125 @@ import Random from "@shared/utils/random.ts";
  *                   3 - hard - choosing often between characters
  * @param random the random object
  */
-export function generateString(length: number, difficulty: number, random: Random): string {
-    if (difficulty === 0) {
-        return ''
-    }
-
-    const tmpFrequencies : {[key: number] : number[][] } = {
-        8: [
-            [1, 2, 5],
-            [1, 3, 4],
-        ],
-        9: [
-            [1, 2, 6],
-            [1, 3, 5],
-            [2, 3, 4],
-        ],
-        10: [
-            [1, 2, 7],
-            [1, 3, 6],
-        ],
-        11: [
-            [1, 2, 8],
-            [1, 3, 7],
-            [1, 4, 6],
-            [2, 4, 5],
-            [2, 3, 6],
-        ],
-        12: [
-            //[1, 2, 9],
-            //[1, 3, 8],
-            //[1, 4, 7],
-            [2, 3, 7],
-            [3, 4, 5],
-            [1, 2, 4, 5],
-            [1, 2, 4, 5],
-        ],
-        13: [
-            //[1, 4, 8],
-            //[1, 5, 7],
-            //[2, 3, 8],
-            [2, 4, 7],
-            //[2, 5, 6],
-            [3, 4, 6],
-            [1, 2, 4, 6],
-            [1, 2, 4, 6],
-        ]
-    }
-
-    if (difficulty === 1) {
-        // generate a word length at maximum of 13 chars
-        const chosenFrequency = tmpFrequencies[length]
-        const randomIndex = random.int(0, chosenFrequency.length-1)
-
-        return generateWordBasedOnFrequency(chosenFrequency[randomIndex], random)
-    }
-    else {
-        // generate a word length with any amount of chars
-        return ''
-    }
-
+export function generateString(
+  length: number,
+  difficulty: number,
+  random: Random,
+): string {
+  if (difficulty === 0) {
+    return ""
+  }
+
+  const tmpFrequencies: { [key: number]: number[][] } = {
```

Comment:
If this dictionary never changes, it could be defined outside of the function scope.

Reply:
The type of `length` could then be `keyof tmpFrequencies`